src/backend/distributed/commands/index.c

```diff
@@ -168,6 +168,67 @@ PlanIndexStmt(IndexStmt *createIndexStatement, const char *createIndexCommand)
 }
 
 
+/*
+ * PlanReindexStmt determines whether a given REINDEX statement involves
+ * a distributed table. If so, error until support is implemented.
+ */
+List *
+PlanReindexStmt(ReindexStmt *ReindexStatement, const char *ReindexCommand)
+{
+	List *ddlJobs = NIL;
+
+	/*
+	 * We first check whether a distributed relation is affected. For that, we need to
+	 * open the relation. To prevent race conditions with later lookups, lock the table.
+	 */
+	if (ReindexStatement->relation != NULL)
+	{
+		Relation relation = NULL;
+		Oid relationId = InvalidOid;
+		bool isDistributedRelation = false;
+		LOCKMODE lockmode = ShareLock;
```

Comment:
I think it should be OK to get `AccessShareLock` here. If Citus won't error out, we'll already go through the `standardProcess_Utility`, which will acquire the accurate locks. Here we're only checking a property of a table, which should be fine with `AccessShareLock`